mobile_verifier/src/rewarder.rs

```diff
@@ -184,96 +231,217 @@ where
+            SpeedtestAverages::aggregate_epoch_averages(reward_period.end, &self.pool).await?;
+        let coverage_points = CoveragePoints::aggregate_points(
+            &self.pool,
+            heartbeats,
+            &speedtest_averages,
+            reward_period.end,
+        )
+        .await?;
+
+        let total_poc_rewards =
+            reward_shares::get_scheduled_tokens_for_poc(reward_period.end - reward_period.start)
+                - transfer_reward_sum;
+
         if let Some(mobile_reward_shares) =
-            coverage_points.into_rewards(transfer_rewards.reward_sum(), reward_period)
+            coverage_points.into_rewards(total_poc_rewards, reward_period)
         {
-            for mobile_reward_share in mobile_reward_shares {
+            // handle poc reward outputs
+            let mut allocated_poc_rewards = 0_u64;
+            for (poc_reward_amount, mobile_reward_share) in mobile_reward_shares {
+                allocated_poc_rewards += poc_reward_amount;
                 self.mobile_rewards
                     .write(mobile_reward_share, [])
                     .await?
                     // Await the returned one shot to ensure that we wrote the file
                     .await??;
             }
+            // write out any unallocated poc reward
+            let unallocated_poc_reward_amount = (total_poc_rewards
+                - Decimal::from(allocated_poc_rewards))
+            .round_dp_with_strategy(0, RoundingStrategy::ToZero)
+            .to_u64()
+            .unwrap_or(0);
+            self.write_unallocated_reward(
+                UnallocatedRewardType::Poc,
+                unallocated_poc_reward_amount,
+                reward_period,
+            )
+            .await?;
+        };
+        Ok(())
+    }
 
-            for mobile_reward_share in transfer_rewards.into_rewards(reward_period) {
-                self.mobile_rewards
-                    .write(mobile_reward_share, [])
-                    .await?
-                    // Await the returned one shot to ensure that we wrote the file
-                    .await??;
-            }
+    async fn reward_dc(
+        &self,
+        reward_period: &Range<DateTime<Utc>>,
+        transfer_rewards: TransferRewards,
+    ) -> anyhow::Result<()> {
+        // handle dc reward outputs
+        let mut allocated_dc_rewards = 0_u64;
+        let total_dc_rewards = transfer_rewards.total();
+        for (dc_reward_amount, mobile_reward_share) in transfer_rewards.into_rewards(reward_period)
+        {
+            allocated_dc_rewards += dc_reward_amount;
+            self.mobile_rewards
+                .write(mobile_reward_share, [])
+                .await?
+                // Await the returned one shot to ensure that we wrote the file
+                .await??;
+        }
+        // write out any unallocated dc reward
+        let unallocated_dc_reward_amount = (total_dc_rewards - Decimal::from(allocated_dc_rewards))
```

Comment:
since unused data rewards get reallocated to the poc pool will there ever be unallocated data rewards?